pointcloud/pointcloud_utils.go

```diff
@@ -69,21 +71,95 @@ func StatisticalOutlierFilter(meanK int, stdDevThresh float64) (func(PointCloud)
 			kd = ToKDTree(pc)
 		}
 		// get the statistical information
+
 		avgDistances := make([]float64, 0, kd.Size())
 		points := make([]PointAndData, 0, kd.Size())
-		kd.Iterate(0, 0, func(v r3.Vector, d Data) bool {
-			neighbors := kd.KNearestNeighbors(v, meanK, false)
-			sumDist := 0.0
-			for _, p := range neighbors {
-				sumDist += v.Distance(p.P)
+		pointAsMatrix, ok := kd.points.(*matrixStorage)
```

Comment:
This is a little type check I had to do because not all kdTrees use matrixStorage